### ISSUE TITLE
Measuring Perplexity (PPL)?

### ISSUE THREAD

**Issue by hotdogstar** [time-series-foundation-models/lag-llama#59]
Hi there! First off, great job on the paper and model. I was wondering if you know a method to measure a forecasted timeseries perplexity? I understand that samples are calculated using the idea from [IQN for Distributional Reinforcement Learning](https://arxiv.org/pdf/1806.06923), but is there still a way to get all probabilities/logits for the next forecast so as to calculate this?

**Comment by ashok-arjun** [comment]
Hi, thanks for the kind words.

We use negative log likelihood as the loss function, so you can basically obtain the negative log likelihood of any `prediction` (by prediction I mean the one-step ahead forecast that we use during the loss; if you want the prediction to mean a multi-step ahead forecast, ensure you set the estimator to the right prediction length before you perform the forecast), when you pass it through our pretrained model. It's basically the [loss](https://github.com/time-series-foundation-models/lag-llama/blob/main/lag_llama/gluon/lightning_module.py#L285). I believe you can calculate the perplexity from this value.

I'm not sure what you mean by `obtaining the probabilities/logits of the next forecast`, as our model does not give discrete outputs. But again, you don't need such probs/logits to compute perplexity.

**Comment by hotdogstar** [comment]
Thanks a lot!
